### PR TITLE
docs: note about firewall rules for GKE private clusters

### PR DESCRIPTION
**Pull request by CyberHippo**
resolves https://github.com/open-telemetry/opentelemetry-operator/issues/1009

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -229,6 +229,8 @@ The possible values for the annotation can be
 
 >**Note:** For `DotNet` auto-instrumentation, by default, operator sets the `OTEL_DOTNET_AUTO_TRACES_ENABLED_INSTRUMENTATIONS` environment variable which specifies the list of traces source instrumentations you want to enable. The value that is set by default by the operator is all available instrumentations supported by the `openTelemery-dotnet-instrumentation` release consumed in the image, i.e. `AspNet,HttpClient,SqlClient`. This value can be overriden by configuring the environment variable explicitely.
 
+>**Note:** For private GKE clusters, you will need to either add a firewall rule that allows master nodes access to port `9443/tcp` on worker nodes, or change the existing rule that allows access to port `80/tcp`, `443/tcp` and `10254/tcp` to also allow access to port `9443/tcp`. More information can be found in the [Official GCP Documentation](https://cloud.google.com/load-balancing/docs/tcp/setting-up-tcp#config-hc-firewall). See the [GKE documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#add_firewall_rules) on adding rules and the [Kubernetes issue](https://github.com/kubernetes/kubernetes/issues/79739) for more detail.
```
Comment (pavolloffay) [inline README.md:232]: Is this relevant only for the auto-instrumentation or collector as well? To me it seems relevant for any operator that uses a webhook. Therefore shall we move it to the top of the readme?

**Comment by jaronoff97** [comment]
@CyberHippo can i reopen this? I just ran in to this in my personal GKE cluster

**Comment by CyberHippo** [comment]
@jaronoff97 Sure go ahead

**Comment by jaronoff97** [comment]
@pavolloffay are we good to merge this? it appears this also adds back a comment about dotnet that was lost in a merge conflict

**Comment by bgou-dr** [comment]
Just hit this as well. Would be great to see this merged and add to the docs.

**Review by yuriolisa — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**